### ISSUE TITLE
result directory missing if backend fails before starting fakemachine.service 

### ISSUE THREAD

**Issue by obbardc** [go-debos/fakemachine#196]
If we error out _before_ faklemachine.service is ran, we get an error about not finding the directory:
```
$ go run cmd/fakemachine/main.go
Running /bin/bash using kvm backend
fakemachine: open /tmp/fakemachine-1054839504/result: no such file or directory
```

Potentially this links in with #150 

full log:
<details><summary>Details</summary>
<p>
```
obbardc@obbardc-t14:~/projects/debos/fakemachine$ go run cmd/fakemachine/main.go  --show-boot
Running /bin/bash using kvm backend
[    0.000000] Linux version 6.6.13-amd64 (debian-kernel@lists.debian.org) (gcc-13 (Debian 13.2.0-10) 13.2.0, GNU ld (GNU Binutils for Debian) 2.41.90.20240115) #1 SMP PREEMPT_DYNAMIC Debian 6.6.13-1 (2024-01-20)
[    0.000000] Command line: console=ttyS0 panic=-1 plymouth.enable=0 systemd.unit=fakemachine.service loglevel=7
[    0.000000] BIOS-provided physical RAM map:
[    0.000000] BIOS-e820: [mem 0x0000000000000000-0x000000000009fbff] usable
[    0.000000] BIOS-e820: [mem 0x000000000009fc00-0x000000000009ffff] reserved
[    0.000000] BIOS-e820: [mem 0x00000000000f0000-0x00000000000fffff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000100000-0x000000007ffdffff] usable
[    0.000000] BIOS-e820: [mem 0x000000007ffe0000-0x000000007fffffff] reserved
[    0.000000] BIOS-e820: [mem 0x00000000feffc000-0x00000000feffffff] reserved
[    0.000000] BIOS-e820: [mem 0x00000000fffc0000-0x00000000ffffffff] reserved
[    0.000000] BIOS-e820: [mem 0x000000fd00000000-0x000000ffffffffff] reserved
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] APIC: Static calls initialized
[    0.000000] SMBIOS 3.0.0 present.
[    0.000000] DMI: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.16.3-debian-1.16.3-2 04/01/2014
[    0.000000] Hypervisor detected: KVM
[    0.000000] kvm-clock: Using msrs 4b564d01 and 4b564d00
[    0.000001] kvm-clock: using sched offset of 1014501661 cycles
[    0.000003] clocksource: kvm-clock: mask: 0xffffffffffffffff max_cycles: 0x1cd42e4dffb, max_idle_ns: 881590591483 ns
[    0.000006] tsc: Detected 1696.818 MHz processor
[    0.000965] last_pfn = 0x7ffe0 max_arch_pfn = 0x400000000
[    0.000996] MTRR map: 4 entries (3 fixed + 1 variable; max 19), built from 8 variable MTRRs
[    0.000999] x86/PAT: Configuration [0-7]: WB  WC  UC- UC  WB  WP  UC- WT  
[    0.003011] found SMP MP-table at [mem 0x000f5440-0x000f544f]
[    0.003044] Using GB pages for direct mapping
[    0.003140] RAMDISK: [mem 0x7f906000-0x7ffdffff]
[    0.003184] ACPI: Early table checksum verification disabled
[    0.003189] ACPI: RSDP 0x00000000000F5270 000014 (v00 BOCHS )
[    0.003193] ACPI: RSDT 0x000000007FFE21F0 000034 (v01 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003214] ACPI: FACP 0x000000007FFE202C 000074 (v01 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003219] ACPI: DSDT 0x000000007FFE0040 001FEC (v01 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003222] ACPI: FACS 0x000000007FFE0000 000040
[    0.003224] ACPI: APIC 0x000000007FFE20A0 0000F0 (v03 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003227] ACPI: HPET 0x000000007FFE2190 000038 (v01 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003229] ACPI: WAET 0x000000007FFE21C8 000028 (v01 BOCHS  BXPC     00000001 BXPC 00000001)
[    0.003231] ACPI: Reserving FACP table memory at [mem 0x7ffe202c-0x7ffe209f]
[    0.003232] ACPI: Reserving DSDT table memory at [mem 0x7ffe0040-0x7ffe202b]
[    0.003233] ACPI: Reserving FACS table memory at [mem 0x7ffe0000-0x7ffe003f]
[    0.003234] ACPI: Reserving APIC table memory at [mem 0x7ffe20a0-0x7ffe218f]
[    0.003234] ACPI: Reserving HPET table memory at [mem 0x7ffe2190-0x7ffe21c7]
[    0.003235] ACPI: Reserving WAET table memory at [mem 0x7ffe21c8-0x7ffe21ef]
[    0.003523] No NUMA configuration found
[    0.003524] Faking a node at [mem 0x0000000000000000-0x000000007ffdffff]
[    0.003531] NODE_DATA(0) allocated [mem 0x7f8db000-0x7f905fff]
[    0.003678] Zone ranges:
[    0.003678]   DMA      [mem 0x0000000000001000-0x0000000000ffffff]
[    0.003680]   DMA32    [mem 0x0000000001000000-0x000000007ffdffff]
[    0.003681]   Normal   empty
[    0.003682]   Device   empty
[    0.003683] Movable zone start for each node
[    0.003685] Early memory node ranges
[    0.003685]   node   0: [mem 0x0000000000001000-0x000000000009efff]
[    0.003687]   node   0: [mem 0x0000000000100000-0x000000007ffdffff]
[    0.003688] Initmem setup node 0 [mem 0x0000000000001000-0x000000007ffdffff]
[    0.003920] On node 0, zone DMA: 1 pages in unavailable ranges
[    0.003938] On node 0, zone DMA: 97 pages in unavailable ranges
[    0.004811] On node 0, zone DMA32: 32 pages in unavailable ranges
[    0.005225] ACPI: PM-Timer IO Port: 0x608
[    0.005241] ACPI: LAPIC_NMI (acpi_id[0xff] dfl dfl lint[0x1])
[    0.005282] IOAPIC[0]: apic_id 0, version 17, address 0xfec00000, GSI 0-23
[    0.005285] ACPI: INT_SRC_OVR (bus 0 bus_irq 0 global_irq 2 dfl dfl)
[    0.005288] ACPI: INT_SRC_OVR (bus 0 bus_irq 5 global_irq 5 high level)
[    0.005289] ACPI: INT_SRC_OVR (bus 0 bus_irq 9 global_irq 9 high level)
[    0.005290] ACPI: INT_SRC_OVR (bus 0 bus_irq 10 global_irq 10 high level)
[    0.005291] ACPI: INT_SRC_OVR (bus 0 bus_irq 11 global_irq 11 high level)
[    0.005295] ACPI: Using ACPI (MADT) for SMP configuration information
[    0.005297] ACPI: HPET id: 0x8086a201 base: 0xfed00000
[    0.005303] TSC deadline timer available
[    0.005304] smpboot: Allowing 16 CPUs, 0 hotplug CPUs
[    0.005325] kvm-guest: APIC: eoi() replaced with kvm_guest_apic_eoi_write()
[    0.005343] kvm-guest: KVM setup pv remote TLB flush
[    0.005347] kvm-guest: setup PV sched yield
[    0.005358] PM: hibernation: Registered nosave memory: [mem 0x00000000-0x00000fff]
[    0.005359] PM: hibernation: Registered nosave memory: [mem 0x0009f000-0x0009ffff]
[    0.005360] PM: hibernation: Registered nosave memory: [mem 0x000a0000-0x000effff]
[    0.005361] PM: hibernation: Registered nosave memory: [mem 0x000f0000-0x000fffff]
[    0.005363] [mem 0x80000000-0xfeffbfff] available for PCI devices
[    0.005364] Booting paravirtualized kernel on KVM
[    0.005367] clocksource: refined-jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645519600211568 ns
[    0.009439] setup_percpu: NR_CPUS:8192 nr_cpumask_bits:16 nr_cpu_ids:16 nr_node_ids:1
[    0.010392] percpu: Embedded 63 pages/cpu s221184 r8192 d28672 u262144
[    0.010453] kvm-guest: PV spinlocks enabled
[    0.010455] PV qspinlock hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.010461] Kernel command line: console=ttyS0 panic=-1 plymouth.enable=0 systemd.unit=fakemachine.service loglevel=7
[    0.010554] random: crng init done
[    0.010945] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.011321] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.011409] Fallback order for Node 0: 0 
[    0.011416] Built 1 zonelists, mobility grouping on.  Total pages: 515808
[    0.011417] Policy zone: DMA32
[    0.011423] mem auto-init: stack:all(zero), heap alloc:on, heap free:off
[    0.013158] Memory: 260860K/2096632K available (16384K kernel code, 2363K rwdata, 10544K rodata, 3960K init, 3628K bss, 90956K reserved, 0K cma-reserved)
[    0.013354] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=16, Nodes=1
[    0.013379] ftrace: allocating 43098 entries in 169 pages
[    0.019656] ftrace: allocated 169 pages with 4 groups
[    0.020268] Dynamic Preempt: voluntary
[    0.020528] rcu: Preemptible hierarchical RCU implementation.
[    0.020529] rcu: 	RCU restricting CPUs from NR_CPUS=8192 to nr_cpu_ids=16.
[    0.020530] 	Trampoline variant of Tasks RCU enabled.
[    0.020530] 	Rude variant of Tasks RCU enabled.
[    0.020531] 	Tracing variant of Tasks RCU enabled.
[    0.020531] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.020532] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=16
[    0.022668] NR_IRQS: 524544, nr_irqs: 552, preallocated irqs: 16
[    0.022886] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.031948] Console: colour VGA+ 80x25
[    0.031992] printk: console [ttyS0] enabled
[    0.139932] ACPI: Core revision 20230628
[    0.141208] clocksource: hpet: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604467 ns
[    0.142811] APIC: Switch to symmetric I/O mode setup
[    0.143994] x2apic enabled
[    0.144798] APIC: Switched APIC routing to: physical x2apic
[    0.145690] kvm-guest: APIC: send_IPI_mask() replaced with kvm_send_ipi_mask()
[    0.146939] kvm-guest: APIC: send_IPI_mask_allbutself() replaced with kvm_send_ipi_mask_allbutself()
[    0.148387] kvm-guest: setup PV IPIs
[    0.150322] ..TIMER: vector=0x30 apic1=0 pin1=2 apic2=-1 pin2=-1
[    0.151797] tsc: Marking TSC unstable due to TSCs unsynchronized
[    0.153163] Calibrating delay loop (skipped) preset value.. 3393.63 BogoMIPS (lpj=6787272)
[    0.157298] x86/cpu: User Mode Instruction Prevention (UMIP) activated
[    0.158901] Last level iTLB entries: 4KB 512, 2MB 255, 4MB 127
[    0.160239] Last level dTLB entries: 4KB 512, 2MB 255, 4MB 127, 1GB 0
[    0.161175] Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
[    0.163135] Spectre V2 : Mitigation: Retpolines
[    0.165160] Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
[    0.167029] Spectre V2 : Spectre v2 / SpectreRSB : Filling RSB on VMEXIT
[    0.168523] Spectre V2 : Enabling Speculation Barrier for firmware calls
[    0.169161] RETBleed: Mitigation: untrained return thunk
[    0.170346] Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
[    0.173165] Speculative Store Bypass: Mitigation: Speculative Store Bypass disabled via prctl
[    0.175123] Speculative Return Stack Overflow: IBPB-extending microcode not applied!
[    0.177164] Speculative Return Stack Overflow: WARNING: See https://kernel.org/doc/html/latest/admin-guide/hw-vuln/srso.html for mitigation options.
[    0.177171] Speculative Return Stack Overflow: Vulnerable: Safe RET, no microcode
[    0.181188] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
[    0.182908] x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
[    0.185162] x86/fpu: Supporting XSAVE feature 0x004: 'AVX registers'
[    0.186559] x86/fpu: xstate_offset[2]:  576, xstate_sizes[2]:  256
[    0.187946] x86/fpu: Enabled xstate features 0x7, context size is 832 bytes, using 'compacted' format.
[    0.217943] Freeing SMP alternatives memory: 36K
[    0.218938] pid_max: default: 32768 minimum: 301
[    0.220006] LSM: initializing lsm=lockdown,capability,landlock,yama,apparmor,tomoyo,bpf,integrity
[    0.221215] landlock: Up and running.
[    0.221993] Yama: disabled by default; enable with sysctl kernel.yama.*
[    0.223527] AppArmor: AppArmor initialized
[    0.225177] TOMOYO Linux initialized
[    0.225931] LSM support for eBPF active
[    0.226859] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.228514] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.229691] smpboot: CPU0: AMD Ryzen 7 PRO 4750U with Radeon Graphics (family: 0x17, model: 0x60, stepping: 0x1)
[    0.233152] RCU Tasks: Setting shift to 4 and lim to 1 rcu_task_cb_adjust=1.
[    0.233286] RCU Tasks Rude: Setting shift to 4 and lim to 1 rcu_task_cb_adjust=1.
[    0.235287] RCU Tasks Trace: Setting shift to 4 and lim to 1 rcu_task_cb_adjust=1.
[    0.237209] Performance Events: Fam17h+ core perfctr, AMD PMU driver.
[    0.238793] ... version:                0
[    0.239745] ... bit width:              48
[    0.240727] ... generic registers:      6
[    0.241172] ... value mask:             0000ffffffffffff
[    0.242547] ... max period:             00007fffffffffff
[    0.243949] ... fixed-purpose events:   0
[    0.245017] ... event mask:             000000000000003f
[    0.245377] signal: max sigframe size: 1776
[    0.246484] rcu: Hierarchical SRCU implementation.
[    0.247560] rcu: 	Max phase no-delay instances is 1000.
[    0.253619] smp: Bringing up secondary CPUs ...
[    0.255081] smpboot: x86: Booting SMP configuration:
[    0.256255] .... node  #0, CPUs:        #1  #2  #3  #4  #5  #6  #7  #8  #9 #10 #11 #12 #13 #14 #15
[    0.289738] smp: Brought up 1 node, 16 CPUs
[    0.292591] smpboot: Max logical packages: 1
[    0.293189] smpboot: Total of 16 processors activated (54298.17 BogoMIPS)
[    0.303340] node 0 deferred pages initialised in 4ms
[    0.305517] devtmpfs: initialized
[    0.306116] x86/mm: Memory block size: 128MB
[    0.309359] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.311931] futex hash table entries: 4096 (order: 6, 262144 bytes, linear)
[    0.313500] pinctrl core: initialized pinctrl subsystem
[    0.317190] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.319436] DMA: preallocated 256 KiB GFP_KERNEL pool for atomic allocations
[    0.321189] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.322986] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.324837] audit: initializing netlink subsys (disabled)
[    0.325216] audit: type=2000 audit(1707916274.297:1): state=initialized audit_enabled=0 res=1
[    0.325445] thermal_sys: Registered thermal governor 'fair_share'
[    0.327301] thermal_sys: Registered thermal governor 'bang_bang'
[    0.328627] thermal_sys: Registered thermal governor 'step_wise'
[    0.329225] thermal_sys: Registered thermal governor 'user_space'
[    0.330449] thermal_sys: Registered thermal governor 'power_allocator'
[    0.331873] cpuidle: using governor ladder
[    0.334074] cpuidle: using governor menu
[    0.335495] acpiphp: ACPI Hot Plug PCI Controller Driver version: 0.5
[    0.337582] PCI: Using configuration type 1 for base access
[    0.338767] PCI: Using configuration type 1 for extended access
[    0.341494] kprobes: kprobe jump-optimization is enabled. All kprobes are optimized if possible.
[    0.346266] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.348095] HugeTLB: 16380 KiB vmemmap can be freed for a 1.00 GiB page
[    0.349165] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.350621] HugeTLB: 28 KiB vmemmap can be freed for a 2.00 MiB page
[    0.353402] ACPI: Added _OSI(Module Device)
[    0.354295] ACPI: Added _OSI(Processor Device)
[    0.355234] ACPI: Added _OSI(3.0 _SCP Extensions)
[    0.357163] ACPI: Added _OSI(Processor Aggregator Device)
[    0.359868] ACPI: 1 ACPI AML tables successfully acquired and loaded
[    0.365271] ACPI: _OSC evaluation for CPUs failed, trying _PDC
[    0.366887] ACPI: Interpreter enabled
[    0.368005] ACPI: PM: (supports S0 S3 S4 S5)
[    0.369184] ACPI: Using IOAPIC for interrupt routing
[    0.370264] PCI: Using host bridge windows from ACPI; if necessary, use "pci=nocrs" and report a bug
[    0.372220] PCI: Using E820 reservations for host bridge windows
[    0.373433] ACPI: Enabled 2 GPEs in block 00 to 0F
[    0.380884] ACPI: PCI Root Bridge [PCI0] (domain 0000 [bus 00-ff])
[    0.381204] acpi PNP0A03:00: _OSC: OS supports [ExtendedConfig ASPM ClockPM Segments MSI HPX-Type3]
[    0.384257] acpiphp: Slot [3] registered
[    0.385214] acpiphp: Slot [4] registered
[    0.386184] acpiphp: Slot [5] registered
[    0.387154] acpiphp: Slot [6] registered
[    0.388108] acpiphp: Slot [7] registered
[    0.389091] acpiphp: Slot [8] registered
[    0.389236] acpiphp: Slot [9] registered
[    0.390297] acpiphp: Slot [10] registered
[    0.391375] acpiphp: Slot [11] registered
[    0.392448] acpiphp: Slot [12] registered
[    0.393217] acpiphp: Slot [13] registered
[    0.394188] acpiphp: Slot [14] registered
[    0.395173] acpiphp: Slot [15] registered
[    0.396148] acpiphp: Slot [16] registered
[    0.397151] acpiphp: Slot [17] registered
[    0.397214] acpiphp: Slot [18] registered
[    0.398299] acpiphp: Slot [19] registered
[    0.399351] acpiphp: Slot [20] registered
[    0.400434] acpiphp: Slot [21] registered
[    0.401220] acpiphp: Slot [22] registered
[    0.402282] acpiphp: Slot [23] registered
[    0.403303] acpiphp: Slot [24] registered
[    0.404340] acpiphp: Slot [25] registered
[    0.405207] acpiphp: Slot [26] registered
[    0.406164] acpiphp: Slot [27] registered
[    0.407153] acpiphp: Slot [28] registered
[    0.408188] acpiphp: Slot [29] registered
[    0.409224] acpiphp: Slot [30] registered
[    0.410208] acpiphp: Slot [31] registered
[    0.411201] PCI host bridge to bus 0000:00
[    0.412297] pci_bus 0000:00: root bus resource [io  0x0000-0x0cf7 window]
[    0.413179] pci_bus 0000:00: root bus resource [io  0x0d00-0xffff window]
[    0.414813] pci_bus 0000:00: root bus resource [mem 0x000a0000-0x000bffff window]
[    0.416358] pci_bus 0000:00: root bus resource [mem 0x80000000-0xfebfffff window]
[    0.417174] pci_bus 0000:00: root bus resource [mem 0x100000000-0x17fffffff window]
[    0.418832] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.420161] pci 0000:00:00.0: [8086:1237] type 00 class 0x060000
[    0.421765] pci 0000:00:01.0: [8086:7000] type 00 class 0x060100
[    0.423692] pci 0000:00:01.1: [8086:7010] type 00 class 0x010180
[    0.428276] pci 0000:00:01.1: reg 0x20: [io  0xc200-0xc20f]
[    0.430579] pci 0000:00:01.1: legacy IDE quirk: reg 0x10: [io  0x01f0-0x01f7]
[    0.432055] pci 0000:00:01.1: legacy IDE quirk: reg 0x14: [io  0x03f6]
[    0.433164] pci 0000:00:01.1: legacy IDE quirk: reg 0x18: [io  0x0170-0x0177]
[    0.435792] pci 0000:00:01.1: legacy IDE quirk: reg 0x1c: [io  0x0376]
[    0.437393] pci 0000:00:01.3: [8086:7113] type 00 class 0x068000
[    0.439594] pci 0000:00:01.3: quirk: [io  0x0600-0x063f] claimed by PIIX4 ACPI
[    0.441179] pci 0000:00:01.3: quirk: [io  0x0700-0x070f] claimed by PIIX4 SMB
[    0.442935] pci 0000:00:02.0: [1234:1111] type 00 class 0x030000
[    0.445728] pci 0000:00:02.0: reg 0x10: [mem 0xfd000000-0xfdffffff pref]
[    0.450902] pci 0000:00:02.0: reg 0x18: [mem 0xfebd0000-0xfebd0fff]
[    0.463787] pci 0000:00:02.0: reg 0x30: [mem 0xfebc0000-0xfebcffff pref]
[    0.465354] pci 0000:00:02.0: Video device with shadowed ROM at [mem 0x000c0000-0x000dffff]
[    0.468948] pci 0000:00:03.0: [1af4:1000] type 00 class 0x020000
[    0.470535] pci 0000:00:03.0: reg 0x10: [io  0xc1c0-0xc1df]
[    0.476273] pci 0000:00:03.0: reg 0x20: [mem 0xfe000000-0xfe003fff 64bit pref]
[    0.478431] pci 0000:00:03.0: reg 0x30: [mem 0xfeb80000-0xfebbffff pref]
[    0.481522] pci 0000:00:04.0: [1af4:1009] type 00 class 0x000200
[    0.484628] pci 0000:00:04.0: reg 0x10: [io  0xc1e0-0xc1ff]
[    0.486803] pci 0000:00:04.0: reg 0x14: [mem 0xfebd1000-0xfebd1fff]
[    0.493181] pci 0000:00:04.0: reg 0x20: [mem 0xfe004000-0xfe007fff 64bit pref]
[    0.500012] pci 0000:00:05.0: [1af4:1009] type 00 class 0x000200
[    0.502674] pci 0000:00:05.0: reg 0x10: [io  0xc000-0xc03f]
[    0.505167] pci 0000:00:05.0: reg 0x14: [mem 0xfebd2000-0xfebd2fff]
[    0.510682] pci 0000:00:05.0: reg 0x20: [mem 0xfe008000-0xfe00bfff 64bit pref]
[    0.516277] pci 0000:00:06.0: [1af4:1009] type 00 class 0x000200
[    0.518738] pci 0000:00:06.0: reg 0x10: [io  0xc040-0xc07f]
[    0.521168] pci 0000:00:06.0: reg 0x14: [mem 0xfebd3000-0xfebd3fff]
[    0.528436] pci 0000:00:06.0: reg 0x20: [mem 0xfe00c000-0xfe00ffff 64bit pref]
[    0.532074] pci 0000:00:07.0: [1af4:1009] type 00 class 0x000200
[    0.534184] pci 0000:00:07.0: reg 0x10: [io  0xc080-0xc0bf]
[    0.536005] pci 0000:00:07.0: reg 0x14: [mem 0xfebd4000-0xfebd4fff]
[    0.539869] pci 0000:00:07.0: reg 0x20: [mem 0xfe010000-0xfe013fff 64bit pref]
[    0.545891] pci 0000:00:08.0: [1af4:1009] type 00 class 0x000200
[    0.547723] pci 0000:00:08.0: reg 0x10: [io  0xc0c0-0xc0ff]
[    0.549171] pci 0000:00:08.0: reg 0x14: [mem 0xfebd5000-0xfebd5fff]
[    0.553174] pci 0000:00:08.0: reg 0x20: [mem 0xfe014000-0xfe017fff 64bit pref]
[    0.558544] pci 0000:00:09.0: [1af4:1009] type 00 class 0x000200
[    0.560924] pci 0000:00:09.0: reg 0x10: [io  0xc100-0xc13f]
[    0.562265] pci 0000:00:09.0: reg 0x14: [mem 0xfebd6000-0xfebd6fff]
[    0.569188] pci 0000:00:09.0: reg 0x20: [mem 0xfe018000-0xfe01bfff 64bit pref]
[    0.573060] pci 0000:00:0a.0: [1af4:1009] type 00 class 0x000200
[    0.574325] pci 0000:00:0a.0: reg 0x10: [io  0xc140-0xc17f]
[    0.577769] pci 0000:00:0a.0: reg 0x14: [mem 0xfebd7000-0xfebd7fff]
[    0.582394] pci 0000:00:0a.0: reg 0x20: [mem 0xfe01c000-0xfe01ffff 64bit pref]
[    0.588054] pci 0000:00:0b.0: [1af4:1009] type 00 class 0x000200
[    0.590740] pci 0000:00:0b.0: reg 0x10: [io  0xc180-0xc1bf]
[    0.593152] pci 0000:00:0b.0: reg 0x14: [mem 0xfebd8000-0xfebd8fff]
[    0.599084] pci 0000:00:0b.0: reg 0x20: [mem 0xfe020000-0xfe023fff 64bit pref]
[    0.602899] ACPI: PCI: Interrupt link LNKA configured for IRQ 10
[    0.604287] ACPI: PCI: Interrupt link LNKB configured for IRQ 10
[    0.605403] ACPI: PCI: Interrupt link LNKC configured for IRQ 11
[    0.606693] ACPI: PCI: Interrupt link LNKD configured for IRQ 11
[    0.609096] ACPI: PCI: Interrupt link LNKS configured for IRQ 9
[    0.610682] iommu: Default domain type: Translated
[    0.610682] iommu: DMA domain TLB invalidation policy: lazy mode
[    0.611327] pps_core: LinuxPPS API ver. 1 registered
[    0.613169] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.615194] PTP clock support registered
[    0.616365] EDAC MC: Ver: 3.0.0
[    0.617557] NetLabel: Initializing
[    0.618318] NetLabel:  domain hash size = 128
[    0.619096] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.620050] NetLabel:  unlabeled traffic allowed by default
[    0.621165] PCI: Using ACPI for IRQ routing
[    0.622099] pci 0000:00:02.0: vgaarb: setting as boot VGA device
[    0.622176] pci 0000:00:02.0: vgaarb: bridge control possible
[    0.623131] pci 0000:00:02.0: vgaarb: VGA device added: decodes=io+mem,owns=io+mem,locks=none
[    0.625172] vgaarb: loaded
[    0.625827] hpet0: at MMIO 0xfed00000, IRQs 2, 8, 0
[    0.626677] hpet0: 3 comparators, 64-bit 100.000000 MHz counter
[    0.632360] clocksource: Switched to clocksource kvm-clock
[    0.632360] VFS: Disk quotas dquot_6.6.0
[    0.632669] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.634403] AppArmor: AppArmor Filesystem Enabled
[    0.634403] pnp: PnP ACPI init
[    0.634403] pnp: PnP ACPI: found 6 devices
[    0.645430] clocksource: acpi_pm: mask: 0xffffff max_cycles: 0xffffff, max_idle_ns: 2085701024 ns
[    0.647537] NET: Registered PF_INET protocol family
[    0.648664] IP idents hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.650936] tcp_listen_portaddr_hash hash table entries: 1024 (order: 2, 16384 bytes, linear)
[    0.653132] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.654861] TCP established hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.656577] TCP bind hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.658119] TCP: Hash tables configured (established 16384 bind 16384)
[    0.659703] MPTCP token hash table entries: 2048 (order: 3, 49152 bytes, linear)
[    0.661258] UDP hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    0.662658] UDP-Lite hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    0.664202] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.665220] NET: Registered PF_XDP protocol family
[    0.665995] pci_bus 0000:00: resource 4 [io  0x0000-0x0cf7 window]
[    0.667011] pci_bus 0000:00: resource 5 [io  0x0d00-0xffff window]
[    0.667987] pci_bus 0000:00: resource 6 [mem 0x000a0000-0x000bffff window]
[    0.669093] pci_bus 0000:00: resource 7 [mem 0x80000000-0xfebfffff window]
[    0.670203] pci_bus 0000:00: resource 8 [mem 0x100000000-0x17fffffff window]
[    0.671454] pci 0000:00:01.0: PIIX3: Enabling Passive Release
[    0.672448] pci 0000:00:00.0: Limiting direct PCI/PCI transfers
[    0.673553] PCI: CLS 0 bytes, default 64
[    0.674462] Trying to unpack rootfs image as initramfs...
[    0.674966] Initialise system trusted keyrings
[    0.677409] Key type blacklist registered
[    0.678466] workingset: timestamp_bits=36 max_order=19 bucket_order=0
[    0.679581] zbud: loaded
[    0.680271] fuse: init (API version 7.39)
[    0.681789] integrity: Platform Keyring initialized
[    0.682659] integrity: Machine keyring initialized
[    0.692429] Freeing initrd memory: 7016K
[    0.694605] Key type asymmetric registered
[    0.695366] Asymmetric key parser 'x509' registered
[    0.698727] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 247)
[    0.700479] io scheduler mq-deadline registered
[    0.703860] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    0.744505] ACPI: \_SB_.LNKC: Enabled at IRQ 11
[    0.786538] ACPI: \_SB_.LNKD: Enabled at IRQ 10
[    0.832280] ACPI: \_SB_.LNKA: Enabled at IRQ 10
[    0.877965] ACPI: \_SB_.LNKB: Enabled at IRQ 11
[    1.061976] Serial: 8250/16550 driver, 4 ports, IRQ sharing enabled
[    1.063453] 00:04: ttyS0 at I/O 0x3f8 (irq = 4, base_baud = 115200) is a 16550A
[    1.065495] Linux agpgart interface v0.103
[    1.066378] AMD-Vi: AMD IOMMUv2 functionality not available on this system - This is not a bug.
[    1.071874] i8042: PNP: PS/2 Controller [PNP0303:KBD,PNP0f13:MOU] at 0x60,0x64 irq 1,12
[    1.074151] serio: i8042 KBD port at 0x60,0x64 irq 1
[    1.075116] serio: i8042 AUX port at 0x60,0x64 irq 12
[    1.076390] mousedev: PS/2 mouse device common for all mice
[    1.077515] rtc_cmos 00:05: RTC can wake from S4
[    1.078996] input: AT Translated Set 2 keyboard as /devices/platform/i8042/serio0/input/input0
[    1.081264] rtc_cmos 00:05: registered as rtc0
[    1.082131] rtc_cmos 00:05: setting system clock to 2024-02-14T13:11:15 UTC (1707916275)
[    1.083640] rtc_cmos 00:05: alarms up to one day, y3k, 242 bytes nvram, hpet irqs
[    1.084942] amd_pstate: the _CPC object is not present in SBIOS or ACPI disabled
[    1.087278] ledtrig-cpu: registered to indicate activity on CPUs
[    1.102212] NET: Registered PF_INET6 protocol family
[    1.105970] Segment Routing with IPv6
[    1.106713] In-situ OAM (IOAM) with IPv6
[    1.107408] mip6: Mobile IPv6
[    1.107916] NET: Registered PF_PACKET protocol family
[    1.108988] mpls_gso: MPLS GSO support
[    1.111691] IPI shorthand broadcast: enabled
[    1.114378] sched_clock: Marking stable (992017580, 122259254)->(1193094122, -78817288)
[    1.116471] registered taskstats version 1
[    1.117842] Loading compiled-in X.509 certificates
[    1.140467] Loaded X.509 cert 'Build time autogenerated kernel key: 98f104c76b3c958d95fe1d6dbec905bb569437e5'
[    1.145393] Key type .fscrypt registered
[    1.146161] Key type fscrypt-provisioning registered
[    1.149057] Key type encrypted registered
[    1.149830] AppArmor: AppArmor sha1 policy hashing enabled
[    1.151061] ima: No TPM chip found, activating TPM-bypass!
[    1.152538] ima: Allocated hash algorithm: sha256
[    1.153418] ima: No architecture policies found
[    1.154372] evm: Initialising EVM extended attributes:
[    1.155381] evm: security.selinux
[    1.155961] evm: security.SMACK64 (disabled)
[    1.156730] evm: security.SMACK64EXEC (disabled)
[    1.157652] evm: security.SMACK64TRANSMUTE (disabled)
[    1.158590] evm: security.SMACK64MMAP (disabled)
[    1.159468] evm: security.apparmor
[    1.160091] evm: security.ima
[    1.160682] evm: security.capability
[    1.161302] evm: HMAC attrs: 0x1
[    1.269845] RAS: Correctable Errors collector initialized.
[    1.271012] clk: Disabling unused clocks
[    1.273788] Freeing unused decrypted memory: 2028K
[    1.275304] Freeing unused kernel image (initmem) memory: 3960K
[    1.306473] Write protecting the kernel read-only data: 28672k
[    1.309297] Freeing unused kernel image (rodata/data gap) memory: 1744K
[    1.372743] x86/mm: Checked W+X mappings: passed, no W+X pages found.
[    1.373793] Run /init as init process
modprobe: can't load module virtio_console (/usr/lib/modules/6.6.13-amd64/kernel/drivers/char/virtio_console.ko): No such file or directory
modprobe: can't load module 9pnet (/usr/lib/modules/6.6.13-amd64/kernel/net/9p/9pnet.ko): No such file or directory
modprobe: can't load module netfs (/usr/lib/modules/6.6.13-amd64/kernel/fs/netfs/netfs.ko): No such file or directory
ernel panic - nousr on /usr failed: No such device[    1.389465] K
t syncing: Attem/inpted to kill init! exitcode=0x00007f00
[    1.390805] CPU: 0 PID: 1 Comm: init Not tainted 6.6.13-amd64 #1  Debian 6.6.13-1
i[    t1: exec: line 2.391857] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.16.3-debian-1.16.3-2 04/01/2014
[    1.393556] Call Trace:
2: /lib/systemd/[    1.393969]  <TASK>
[    1.394517]  dump_stack_lvl+0x47/0x60
systemd: not found
[    1.395083]  panic+0x180/0x330
[    1.395836]  do_exit+0x98d/0xb00
[    1.396338]  do_group_exit+0x31/0x80
[    1.396906]  __x64_sys_exit_group+0x18/0x20
[    1.397547]  do_syscall_64+0x60/0xc0
[    1.398106]  ? srso_return_thunk+0x5/0x10
[    1.398724]  ? count_memcg_events.constprop.0+0x1a/0x30
[    1.399540]  ? srso_return_thunk+0x5/0x10
[    1.400157]  ? handle_mm_fault+0xa2/0x360
[    1.400794]  ? srso_return_thunk+0x5/0x10
[    1.401414]  ? do_user_addr_fault+0x30f/0x660
[    1.402088]  ? srso_return_thunk+0x5/0x10
[    1.402717]  ? exit_to_user_mode_prepare+0x40/0x1e0
[    1.403467]  entry_SYSCALL_64_after_hwframe+0x6e/0xd8
[    1.404254] RIP: 0033:0x459635
[    1.404758] Code: 89 01 48 83 c8 ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 40 00 48 c7 c6 e0 ff ff ff ba e7 00 00 00 eb 03 66 90 f4 89 d0 0f 05 <48> 3d 00 f0 ff ff 76 f3 f7 d8 64 89 06 eb ec 66 2e 0f 1f 84 00 00
[    1.407578] RSP: 002b:00007ffec1aed718 EFLAGS: 00000202 ORIG_RAX: 00000000000000e7
[    1.408764] RAX: ffffffffffffffda RBX: 00007ffec1aeef04 RCX: 0000000000459635
[    1.409855] RDX: 00000000000000e7 RSI: ffffffffffffffe0 RDI: 000000000000007f
[    1.410954] RBP: 0000000000000003 R08: 00007ffec1aed800 R09: 00000000000000bc
[    1.412037] R10: 0000000000000000 R11: 0000000000000202 R12: 00007ffec1aedbc8
[    1.413143] R13: 00000000005da6a8 R14: 0000000000000000 R15: 0000000000000001
[    1.414212]  </TASK>
[    1.414955] Kernel Offset: 0x3b600000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
fakemachine: open /tmp/fakemachine-1291763863/result: no such file or directory
exit status 255
```
</p>
</details>

**Comment by NeonDaniel** [comment]
I believe I've observed this error running `debos` both via container and local installation in an Ubuntu VM, but running `fakemachine` in a terminal does work. Is there any solution or workaround for this?

**Comment by obbardc** [comment]
This can be reproduced with:

```
diff --git a/machine.go b/machine.go
index 1a627ad..59f07ae 100644
--- a/machine.go
+++ b/machine.go
@@ -346,7 +346,7 @@ OnFailure=poweroff.target
 [Service]
 Environment=HOME=/root IN_FAKE_MACHINE=yes %[2]s
 WorkingDirectory=-/scratch
-ExecStart=/wrapper
+ExecStart=/non_existent_file
 ExecStopPost=/bin/sync
 ExecStopPost=/bin/systemctl poweroff -q -ff
 Type=idle
```

**Comment by obbardc** [comment]
This should have been fixed in 1cb07bb5932f175283219a2ced6dbfa566575ca
